docs/Contributing-back.rst

```diff
@@ -0,0 +1,610 @@
+
+Merging of pull requests
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Once your pull request have been given the green light by Travis,
+and one or more people reviewing have approved the changes, it can be merged into
+the ``develop`` branch. This can only be done by someone in the EasyBuild
+'*admin*' team.
+
+Usually, merging a pull request implies that the changes will be part of the next EasyBuild release.
+
+
+.. _contributing_back_review_process:
+
+Review process for contributions
+--------------------------------
+
+Each contribution is thoroughly reviewed and tested before it gets merged in.
+Some aspects of this are fully automated, others require human intervention.
+
+It is important to be aware of all of the aspects of the review process,
+since **only contributions that fulfill all of the requirements discussed below
+are eligible for being merged in**:
+
+* unit tests must still pass (see :ref:`contributing_back_review_process_travis`)
+
+  * more tests must be added when appropriate (see :ref:`contributing_back_review_process_adding_tests`)
+
+* backward compatibility should be retained (see :ref:`contributing_back_review_process_backward_compatibility`)
+* code style must be kept consistent (see :ref:`contributing_back_review_process_code_style`)
+
+  * easyconfigs should be kept consistent across versions & toolchains (see :ref:`contributing_back_review_process_review_pr`)
+
+* test reports must be submitted for easyconfig contributions (see :ref:`contributing_back_review_process_test_reports`)
+
+
+.. _contributing_back_review_process_travis:
+
+Test suite (Travis)
+~~~~~~~~~~~~~~~~~~~
+
+Each pull request is tested automatically by Travis, and the test result
+is reported in the pull request.
+
+**Only pull requests that have been tested and approved by Travis are
+eligible for being merged!**
+
+Note that Travis will only run the *unit test suite* for that particular repository.
+That is, for easyconfig contributions it does *not* include
+actually building and installing software.
+
+.. _contributing_back_review_process_adding_tests:
+
+Adding tests
+++++++++++++
+
+An implicit requirement for contributions, in particular contributions to the EasyBuild framework,
+is that they **must be accompanied by additional tests or test cases**.
+
+For new features or enhancements, a dedicated test (case) must be added
+which verifies that the feature implementation works as expected.
+
+For bug fixes, a test (case) must be added that triggers the code path where the bug manifested,
+and which verifies that the bug was indeed fixed.
+
+Tests not only confirm that the implementation is correct, it also helps to ensure
+that any future changes will not affect semantics.
+
+
+.. _contributing_back_review_process_backward_compatibility:
+
+Backward compatibility
+~~~~~~~~~~~~~~~~~~~~~~
+
+**Contributions should retain backward compatibility**, i.e., they should *not*
+make any changes that alter the (default) semantics of the existing code base.
```

Comment:
I don't think semantics is the right word here either. I think interface is probably better. Semantics implies a much more rigorous interpretation of the code in its context. I think backwards compatibility is actually clearer here than the explanation.